### ISSUE TITLE
[Suggestion] Driver building

### ISSUE THREAD

**Issue by RicardoMonteiroSimoes** [annex4-inc/vscode-control4-ext#1]
Hello,

I've been watching this repository with great interest, as I'm also starting to develop drivers and the lackluster DriverEditor isn't efficient. I'm using the current C4 Plugin for VSC which already helps, but this seems like such a big step up! 

I noticed that you're integrating the driver editor directly etc. While this is completely fine, recently I've stumbled upon this:
https://github.com/control4/drivers-driverpackager

This is a collection of python scripts to directly create the driver without the need for the driver editor. The question would be, do you think this could be, somehow, integrated into the plugin? Or would you argue that this is out of scope?

Either way, many thanks for the work you've already done!

**Comment by annex4** [comment]
At this point if OpenSSL is installed on the system you can now use that without requiring Driver Editor or DriverPackager.exe. I'm not against adding support for the python script as a build option, but as there is now already an alternative to DriverPackager.exe via OpenSSL a python option is not high on my priority list.

This should be relatively easy to add through a new build stage and a slight modification to the builder:
https://github.com/annex4-inc/vscode-control4-ext/blob/master/client/src/control4/builder.ts

The OpenSSL stage or Manifest stage should act as good examples if you want to try and implement it. The focus right now is making sure the UI capability section is working properly before finishing the roadmap. Once that's done I can come back and take a look at a python build stage.

**Comment by zhouji0212** [comment]
can u show more c4 project examples?

**Comment by annex4** [comment]
Yes, I'll provide more examples once I'm nearing a full release. Is there a specific proxy or type of driver you're looking for an example of?

**Comment by zhouji0212** [comment]
thanks, i want some Curtain AirCondition example that use tcp communication. i want to know how proxy setup

**Comment by zhouji0212** [comment]
now I want a ui button or webview demo.in your camera demo I found the hook function I can't understand.

**Comment by zhouji0212** [comment]
do u have any document about control4 remote debug? I can't find in control4 dealer website or forum

**Comment by annex4** [comment]
I believe Control4 removed the remote debugging API or deprecated it. I never bothered using it so can't help you there.

As for the hook stuff you're talking about in the example, it's pulled from one of our internal libraries and merged into script at build time. It basically just wraps the entry points (like OnDriverInit) into an object that each library can use independently without having to call them all from the main file. Lets me include a library and have it just function without doing anything specific for that driver.

I'll be releasing most of our internal libraries when our new infrastructure comes out in October, as well as wrapping up this extension.

@zhouji0212 If you have further questions please create a new issue for them specifically.
